### PR TITLE
Simplify client types

### PR DESCRIPTION
**Pull request by neongreen**
This PR gets rid of `?`, replacing it with explicit schema manipulation.

This also means that now FromJSON instances will accept missing fields as `Nothing` – aeson has built-in logic for treating `Maybe` fields like this, but it doesn't know anything about `Maybe ... ? ...`.

### CODE REVIEW

File: back/guide.cabal [hunk 1]
```diff
@@ -96,10 +96,7 @@ library
     Guide.Routes
     Imports
     To
-  build-depends:       Spock
```
Comment (neongreen) [inline back/guide.cabal:99]: Ignore these changes – it's churn caused by automatic sorting.

File: back/src/Guide/Api/Types.hs [hunk 2]
```diff
@@ -800,13 +944,5 @@ instance ToSchema (Uid Trait) where
 
 instance ToSchema CategoryStatus
 
-instance ToSchema ItemKind where
```
Comment (neongreen) [inline back/src/Guide/Api/Types.hs:803]: This instance wasn't needed anymore.

**Comment by willbasky** [comment]
Why did you exclude tags?

**Comment by neongreen** [comment]
What do you mean by "tags"?

If you mean "strings after `?`", then they are still available in Swagger, they just got moved into `ToSchema` instances.

**Review by willbasky — APPROVED** [review]

**Pull request merged**